### DISCUSSION TITLE
backend.ai-webui #1131 issue 해결방안

### DISCUSSION THREAD

**Discussion by TeusEE** [lablup/contribution-academy-2021#110] [Q&A]
https://github.com/lablup/backend.ai-webui/issues/1131

해당 이슈 해결을 위해서
1. 프론드엔드 단의 수정으로 해결
2. 백엔드 단의 수정으로 해결

두가지 방법이 모두 가능할것 같아 보입니다.

현재 백앤드 단에 backend.ai/manager/src/ai/backend/manager/registry.py를 수정하는 방법을 생각하고 있습니다.

프론트엔드 단의 수정으로 해결할 수 있는 방법은 어떻게해야할까요?

**Comment by lizable** [comment]
@TeusEE 
I tried the first method by simply removing two thing, [`isGPURequired`](https://github.com/lablup/backend.ai-webui/blob/5b8cfb36dcf047033c099251959ef06ba625364c/src/components/backend-ai-environment-list.ts#L418-L425) code blocks and `fgpu`, `gpu`, `cuda.shares`, `cuda.device` from [imageResource](https://github.com/lablup/backend.ai-webui/blob/5b8cfb36dcf047033c099251959ef06ba625364c/src/components/backend-ai-environment-list.ts#L445) instance, which is the argument of the install method in ContainerImage Class.
 and It seems like downloading itself it OK but, after downloading and I cannot create the session using it.
I got this kind of error...(I tried to download TF 2.5 in stable project in harbor)
![image](https://user-images.githubusercontent.com/46954439/136652183-1a718122-8493-491a-8a23-58f140ef3026.png)

I'll investigate it more.

**Reply by lizable** [reply]
And I also found a logic error in [here](https://github.com/lablup/backend.ai-webui/blob/5b8cfb36dcf047033c099251959ef06ba625364c/src/components/backend-ai-environment-list.ts#L427-L438) as well, so whether this issue cannot be handled in WebUI, it needs to be solved.
